### PR TITLE
Fix scrollToObject: inconsistency bug when scrolling to bottom/right with content inset

### PR DESCRIPTION
**Pull request by qhhonx**
The implementation of 
```Objective-C
-[IGListAdapter scrollToObject:supplementaryKinds:scrollDirection:scrollPosition:animated:]
``` 
has little inconsistencies where the `UICollectionViewScrollPositionLeft` and `UICollectionViewScrollPositionTop` considering the content inset left/top of the collection view was being applied to the final offset. However, the `UICollectionViewScrollPositionRight` and `UICollectionViewScrollPositionBottom` ignoring the content inset right/bottom of the collection view was being applied to the final offset. The different result is that scrolling to the most `Left/Top`, the first section content always be visible by considering its content inset while scrolling to the most `Right/Bottom`, the last section content always be fully displayed but not be fully visible by considering its content inset.

## Changes in this pull request

Issue fixed: #

### Checklist

- [x] All tests pass. Demo project builds and runs.
- [x] I added tests, an experiment, or detailed why my change isn't tested.
- [x] I added an entry to the `CHANGELOG.md` for any breaking changes, enhancements, or bug fixes.
- [x] I have reviewed the [contributing guide](https://github.com/Instagram/IGListKit/blob/master/.github/CONTRIBUTING.md)

### CODE REVIEW

File: Tests/IGListAdapterTests.m [hunk 1]
```diff
@@ -751,13 +751,36 @@ - (void)test_whenScrollVerticallyToBottom_withContentInsets_thatBottomFlushWithC
...
 
     // top 50, bottom 100
     self.collectionView.contentInset = UIEdgeInsetsMake(50, 0, 100, 0);
     [self.collectionView layoutIfNeeded];
     [self.adapter scrollToObject:@100 supplementaryKinds:nil scrollDirection:UICollectionViewScrollDirectionVertical scrollPosition:UICollectionViewScrollPositionBottom animated:NO];
-    IGAssertEqualPoint([self.collectionView contentOffset], 0, 900);
+    IGAssertEqualPoint([self.collectionView contentOffset], 0, 900 + 100);
+    
+    UICollectionViewLayout *layout = self.collectionView.collectionViewLayout;
+    if ([layout isKindOfClass:[UICollectionViewFlowLayout class]]) {
```
Comment (rnystrom) [inline Tests/IGListAdapterTests.m]: Can you change this to a hard cast so the test crashes if it changes?

File: Tests/IGListAdapterTests.m [hunk 2]
```diff
@@ -751,13 +751,36 @@ - (void)test_whenScrollVerticallyToBottom_withContentInsets_thatBottomFlushWithC
...
+        [self.collectionView layoutIfNeeded];
+        [self.adapter scrollToObject:@100 supplementaryKinds:nil scrollDirection:UICollectionViewScrollDirectionHorizontal scrollPosition:UICollectionViewScrollPositionRight animated:NO];
+        IGAssertEqualPoint([self.collectionView contentOffset], 900 + 100, 0);
+        
+        // left 50, right 100
+        self.collectionView.contentInset = UIEdgeInsetsMake(0, 50, 0, 100);
+        [self.collectionView layoutIfNeeded];
+        [self.adapter scrollToObject:@100 supplementaryKinds:nil scrollDirection:UICollectionViewScrollDirectionHorizontal scrollPosition:UICollectionViewScrollPositionRight animated:NO];
+        IGAssertEqualPoint([self.collectionView contentOffset], 900 + 100, 0);
+    }
```
Comment (rnystrom) [inline Tests/IGListAdapterTests.m]: Love tests!!

Can we move this to its own unit test though? That way if it starts failing we can ID the issue faster.

File: Source/IGListAdapter.m [hunk 3]
```diff
@@ -263,7 +263,7 @@ - (void)scrollToObject:(id)object
         case UICollectionViewScrollDirectionHorizontal: {
             switch (scrollPosition) {
                 case UICollectionViewScrollPositionRight:
-                    contentOffset.x = offsetMax - collectionViewWidth - contentInset.left;
+                    contentOffset.x = offsetMax - collectionViewWidth + contentInset.right;
```
Comment (rnystrom) [inline Source/IGListAdapter.m]: wow good find!

File: Tests/IGListAdapterTests.m (hunk 1 above)
Comment (xohozu) [reply Tests/IGListAdapterTests.m]: fixed in 7ed9b9757140f235b5dc9751d94b394b12536033

File: Tests/IGListAdapterTests.m (hunk 2 above)
Comment (xohozu) [reply Tests/IGListAdapterTests.m]: fixed in 7ed9b9757140f235b5dc9751d94b394b12536033

**Comment by xohozu** [comment]
Hi, @rnystrom , I can not figure out why the travis-ci task failed while my local building and testing are passed. Could you please check whether some common mistakes occurred?

**Comment by qhhonx** [comment]
> [@qhhonx](https://github.com/qhhonx) has updated the pull request. [Re-import the pull request](https://our.intern.facebook.com/intern/opensource/github/pull_request/confirm/import/1926981544263367/)

@rnystrom Can you reimport this PR again? It just fixes testcase inconsistency without extra logic code.

**Comment by qhhonx** [comment]
@lorixx It seems that you are active maintainer now. Can you review this PR and import it again?

**Pull request closed without merging**